### DISCUSSION TITLE
Issue with paths when composing configs

### DISCUSSION THREAD

**Discussion by dagap** [facebookresearch/hydra#2394] [Q&A]
I am using hydra where I have my ML experiments organised by projects as follows:

```
----->  experiments
        ------> project
               ------> base.yaml
               ------> dev.yaml
               ------> prod.yaml
```

Now, say the base has something as follows:

```
# experiments/project/base
defaults:
  - override /model: mymodel.yaml
```

Now, this works

```
# experiments/project/dev
defaults:
  - project/base@_here_
```

However, now if I try and do something like this and "inherit" from the `dev`config as:

```
# experiments/project/prod
defaults:
  - project/dev@_here_
```

This runs into `Could not load 'experiments/project/project/base`. So it seems that the paths are not being extrapolated as desired due to this directory nesting. I tried it without the `_here_`clause but runs into the same issue. 

So, at the moment I can only inheit from the `base`config and none of the derived configs. I can work around it by flattening the directories but it would be nice to organise the configs by projects and customers.

**Comment by dagap** [comment]
Managed to sort it out. Required the full path as:

```
/experiments/project/dev@_here_
```

**Comment by Jasha10** [comment]
For yaml files in the same directory, should be able to use just `dev` instead of `/experiments/projects/dev@_here_`:
```yaml
# experiments/project/dev
defaults:
  - base
```
```yaml
# experiments/project/prod
defaults:
  - dev
```

**Reply by dagap** [reply]
@Jasha10 So this is rather weird. If I have the directory structure as above and I do what you suggested, I get:

```
'experiments/project/dev': Could not load 'experiments/base'.
```

I am calling it as `python train.py experiments=project/dev` So it seems it is skipping the `project` subdirectory. This works when the structure is just:

```
experiments
    -> base.yaml
    -> dev.yaml
    -> prod.yaml
```
But if I have subdirectories, it barfs.

**Reply by Jasha10** [reply]
Using a slash on the right side of the equals sign, e.g.`experiments=project/dev`, is not supported by Hydra. You should use `experiments/project=dev`, or something like `experiments/project@experiments=dev`.

**Reply by dagap** [reply]
@Jasha10 very interesting. Thank you for sharing.
